lib/App/MintTag.pm

```diff
@@ -626,8 +626,33 @@ sub _diagnostic_merge ($self, $mrs) {
   }
 
   # If we are in this sub at all, we expect that the above will fail. If it
-  # doesn't, something very strange indeed has happened.
-  $Logger->log_fatal('diagnostic merge succeeded somehow...this should not happen!');
+  # doesn't, something strange has happened.
+  $Logger->log('diagnostic merge succeeded somehow, so here are multiply-affected files');
+
+  my %affected_by;
+  for my $mr (@$mrs) {
+    $Logger->log([ "checking files affected by %s", $mr->ident ]);
+
+    try {
+      my @files = $mr->affected_files;
+      $affected_by{$_}{ $mr->ident } = $mr for @files;
+    } catch {
+      my $err = $_;
+      chomp $err;
+
+      $Logger->log([
+        "error fetching files affected by %s: %s", $mr->ident, $err
+      ]);
+    };
+  }
+
+  for my $file (sort keys %affected_by) {
+    next unless $affected_by{$file}->%* > 1;
```

Comment:
Should this be `next unless keys $affected_by{$file}->%* > 1;` because we don't want to report a file if it's only affected 1 MR, but this would be in list context so (k, v) tuples which will always at least 2?

Reply:
Short answer:  no, this is correct.

Long answer:  What you described is not how this works.  The `>` operator imposes scalar context.  A hash in scalar context is the number of entries.  (This changed in v5.26.0.  Prior to that, a hash in scalar context was Some Useless Garbage.  Specifically, it was a string describing the bucket usage and count.  This is roughly never useful, so `scalar %xyz` became entry count (same as `scalar keys %zys`).  Bucket information is now available in Hash::Util::bucket_X functions.

But the fundamental thing is that `>` imposes scalar context and never operates on lists.  ` (10,9,8,6,5,4,3,2,1) > 2` is false, not true, because `>` has imposed scalar context on the comma operators to its left.

Reply:
```
perl -E '%h = (1..10); say "more than 4" if %h > 4; say "more than 5" if %h > 5'
```

Reply:
Ok right, that all makes sense. Was thinking about scalar context for a moment there.